currency-rate/cmd/main.go

```diff
@@ -1,113 +1,180 @@
+
+	return conn.LocalAddr().String()
 }
 
-func main() {
+func main() { // nolint: funlen
 	if err := settings.InitSettings(); err != nil {
 		slog.Error("failed to initialize settings", slog.Any("error", err))
-		// panic(err)
 	}
+	appConfig = appCfg.NewFromEnv()
+	slog.SetDefault(NewLogger())
+	slog.Debug("settings initialized")
 
-	db, err := InitDatabase()
+	db, err := NewDatabase()
 	if err != nil {
-		slog.Error("failed to initialize database", slog.Any("error", err))
+		slog.Error("failed to create database", slog.Any("error", err))
 		panic(err)
 	}
+	defer db.Close()
 
 	// Initialize rate fetcher chain of responsibilities
-	rateFetcher := InitFetchers()
+	rateService := service.NewRateService(NewFetchers()...)
 
+	transportConfig := transportCfg.NewFromEnv()
+	// Initializer user event producer
+	userProducer, err := userBroker.NewProducer(transportCfg.Config{
+		BrokerURI: transportConfig.BrokerURI,
+		QueueName: appConfig.UserQueueName,
+	})
+	if err != nil {
+		slog.Error("failed to create user producer", slog.Any("error", err))
+		panic(err)
+	}
+	userConsumer, err := userBroker.NewConsumer(transportCfg.Config{
+		BrokerURI: transportConfig.BrokerURI,
+		QueueName: appConfig.UserCompensateQueueName,
+	})
+	if err != nil {
+		slog.Error("failed to create user consumer", slog.Any("error", err))
+		panic(err)
+	}
 	userRepo := models.NewUserRepository(db)
+
+	// Decorate userRepo with userProducer
+	decoratedUserRepo := handler.NewUserRepositorySaga(userRepo, userProducer, userConsumer)
+	slog.Debug("user repository saga created")
+
 	apiClient := server.Client{
 		Config:      serverCfg.NewFromEnv(),
-		RateService: service.NewRateService(models.NewRateRepository(db), rateFetcher),
-		UserRepo:    userRepo,
-	}
-
-	// Start cron job for notifications
-	cronSpec := os.Getenv("CRON_SPEC")
-	if cronSpec == "" {
-		slog.Warn(
-			"CRON_SPEC is not set, using default value",
-			slog.Any("default", defaultCronSpec),
-		)
-		cronSpec = defaultCronSpec
+		RateService: rateService,
+		UserRepo:    decoratedUserRepo,
 	}
 
-	mailerFacade, err := mail.NewMailerFacade(transportCfg.NewFromEnv())
-	if err != nil {
-		slog.Error("failed to initialize mailer facade", slog.Any("error", err))
+	cronManager := NewCron(transportConfig, rateService)
+	if cronManager == nil {
+		slog.Error("failed to initialize cron manager")
+	} else {
+		cronManager.Start()
+		defer cronManager.Stop()
 	}
-	defer mailerFacade.Close()
-	notifier := notifications.NewUsersNotifier(
-		mailerFacade,
-		apiClient.RateService,
-		userRepo,
-		&message.PlainRate{},
-	)
-	StartCron(cronSpec, func() {
-		ctx, cancel := context.WithTimeout(context.Background(), server.RateTimeout)
-		defer cancel()
-		notifier.Notify(ctx)
-	})
 
 	// Start HTTP server
 	s := server.NewServer(apiClient.Config, server.NewEngine(apiClient))
 	slog.Info("starting server", slog.Any("address", s.Addr))
+
+	err = metrics.InitPush(
+		appConfig.VictoriaMetricsPushURL,
+		10*time.Second,
```

Comment:
this parameter is a scrape interval, right? Why don't you put in into the configuration or at least local constant?